src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy

```diff
@@ -122,7 +123,15 @@ class DockerJavaApplicationPlugin implements Plugin<Project> {
         project.task(PUSH_IMAGE_TASK_NAME, type: DockerPushImage) {
             description = 'Pushes created Docker image to the repository.'
             dependsOn dockerBuildImageTask
-            conventionMapping.imageName = { dockerBuildImageTask.getTag() }
+            conventionMapping.imageName = {
+                if (dockerBuildImageTask.getTags().size() > 1) {
```

Comment:
This check forces you to override settings for dockerPushImage iff you specified multiple tags in dockerBuildImage.
This is somewhat ugly, but it's just here because DockerPushImage doesn't support multiple tags to be pushed. Otherwise, the default behavior could be to push all built tags.

The change is not breaking because it behaves exactly the same in the default case of just 1 tag.